### ISSUE TITLE
arrow-body-style fix is overzealous for `as-needed`

### ISSUE THREAD

**Issue by philip-peterson** [eslint/eslint#12092]
**Tell us about your environment**

* **ESLint Version:**
6.1.0

* **Node Version:**
11.10.1

* **yarn Version:**
1.16.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  extends: ['plugin:react-perf/all'],
  plugins: [],
  parserOptions: {
    ecmaVersion: 2016,
    sourceType: 'module',
  },
  env: {
    es6: true,
  },
  globals: {},
  rules: {
    'arrow-body-style': ['warn', 'as-needed'],
    'no-console': 0,
  },
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I have a file like so, and want to use eslint fix on it:

```js
import React from 'react';

const x = () => {
  return (
    <div>foo</div>
  );
};

console.log(x);
```

```bash
yarn fix # which gets translated to eslint src/. --ext .ts,.js,.tsx,.jsx --fix
```

**What did you expect to happen?**
Nothing; the code should pass the lint rule just fine.

**What actually happened? Please include the actual, raw output from ESLint.**
The eslint fixer removed the `return` statement and replaced it with a non-bracketed implicit return statement.  This is the file after the fix:

```js
import React from 'react';

const x = () => (
    <div>foo</div>
  );

console.log(x);
```

**Are you willing to submit a pull request to fix this bug?**
No, sorry. However, I did create a repository for easy reproducing of the bug [https://github.com/philip-peterson/repro-eslint-bug](https://github.com/philip-peterson/repro-eslint-bug).

**Comment by ljharb** [comment]
Why is this overzealous? JSX resolves to a single expression value; an arrow function that returns an expression should use the concise form when using "as-needed".

**Comment by philip-peterson** [comment]
Fair point. Maybe it should work this way.
